src/lib/init.js

```diff
@@ -36,93 +35,54 @@ function readExternalConsentData(config) {
-		[
-			readVendorConsentCookie(),
-			fetchPubVendorList(),
-			readPublisherConsentCookie()
-		]
-	]);
-}
-
-function readGlobalAndExternalConsentData(config) {
-	return Promise.all([
-		readExternalConsentData(config),
-		readVendorConsentCookie()
-	]);
-}
-
-export function init(configUpdates) {
+export function init (configUpdates) {
 	config.update(configUpdates);
 	log.debug('Using configuration:', config);
 
 	// Fetch the current vendor consent before initializing
-	return ((config.getConsentData) ? readGlobalAndExternalConsentData(config) : readInternalConsentData())
-		.then(([[vendorConsentData, pubVendorsList, publisherConsentData], globalVendorConsentData]) => {
-			const {vendors} = pubVendorsList || {};
-
-			// Check config for allowedVendorIds then the pubVendorList
-			const {allowedVendorIds: configVendorIds} = config;
-			const allowedVendorIds = configVendorIds instanceof Array && configVendorIds.length ? configVendorIds :
-				vendors && vendors.map(vendor => vendor.id);
-
-			// Initialize the store with all of our consent data
+	return ((config.getConsentData) ? readExternalConsentData(config) : readConsentCookie())
+		.then((consentData) => {
 			const store = new Store({
 				cmpVersion: CMP_VERSION,
 				cmpId: CMP_ID,
 				cookieVersion: COOKIE_VERSION,
-				vendorConsentData,
-				globalVendorConsentData,
-				publisherConsentData,
-				pubVendorsList,
-				allowedVendorIds
+				consentData,
 			});
 
-			// Pull queued command from __cmp stub
-			const {commandQueue = []} = window[CMP_GLOBAL_NAME] || {};
-
 			// Replace the __cmp with our implementation
-			const cmp = new Cmp(store);
+			const tcfManager = new Cmp(store);
+
+			config.decoratePageCallHandler(CmpApi);
+			const cmpApi = new CmpApi(CMP_ID, CMP_VERSION, createCommands(store, tcfManager));
+			config.setPageCallHandlerContext(cmpApi);
 
-			// Expose `processCommand` as the CMP implementation
-			window[CMP_GLOBAL_NAME] = cmp.processCommand;
+			store.setCmpApi(cmpApi);
 
 			// Notify listeners that the CMP is loaded
 			log.debug(`Successfully loaded CMP version: ${pack.version}`);
-			cmp.isLoaded = true;
-			cmp.notify('isLoaded');
+			tcfManager.isLoaded = true;
+			tcfManager.notify('isLoaded');
 
 			// Render the UI
 			const App = require('../components/app').default;
-			render(<App store={store} notify={cmp.notify} />, document.body);
-
-			// Execute any previously queued command
-			cmp.commandQueue = commandQueue;
-			cmp.processCommandQueue();
+			render(<App store={store} notify={tcfManager.notify} />, document.body);
 
 			let isConsentToolShowing = store.isConsentToolShowing;
 			store.subscribe(store => {
 				if (store.isConsentToolShowing !== isConsentToolShowing) {
 					isConsentToolShowing = store.isConsentToolShowing;
-					cmp.notify('onToggleConsentToolShowing', isConsentToolShowing);
+					// tcfManager.notify('onToggleConsentToolShowing', isConsentToolShowing);
+					isConsentToolShowing && config.onConsentToolShowing();
```

Comment:
Is it technically possible to send custom event here instead calling external function from configuration? It would be better to use events otherwise we will have to pass separate function in `config` object for every such notification. It could make total disaster in code ;)

Reply:
Fixed.